### PR TITLE
docs(breaking-changes): add breaking change policy

### LINKED ISSUE #695: RFC: breaking changes and design tokens
# Context 

Following the addition of a design token for focus styles (see #685), we discussed whether it could be considered a breaking change or not.

Since we introduced a theme that could be customized by people using our components, it does mean that when we release new tokens, people may have to update their own theme.

In the end it seems we don't think it's a breaking change because it adds a new way to customize how clever components look / behave. If you don't customize it, you get a black outline so this does not really break anything but it does look a lot different than before :thinking:.

# Things to discuss 

* Discuss and write down our criteria for breaking changes in a specific document.
* Add a way to warn developers about theme updates.

### PR DESCRIPTION
**Pull request by florian-sanders-cc**
Fixes #695

### CODE REVIEW

File: docs/guides/breaking-change-policy.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+* slots,
+* events,
+* CSS custom properties,
+* methods,
+* host `display` property.
+
+Therefore, the following should be considered breaking changes:
+
+* renaming or deleting one of those elements,
+* modifying the type of an attribute or property and the signature of an event or method.
```
Comment (pdesoyres-cc) [inline docs/guides/breaking-change-policy.md:34]:
```suggestion
* modifying the type of an attribute or property,
* modifying the signature of a public method,
* modifying the type of the object included in an event.
```

**Review by pdesoyres-cc — COMMENTED** [review]
GG, very nice doc.

File: docs/guides/breaking-change-policy.md (hunk 1 above)
Comment (roberttran-cc) [reply docs/guides/breaking-change-policy.md]: Thanks for the suggestion: doc updated!

**Review by pdesoyres-cc — APPROVED** [review]
Perfect

**Pull request merged**